sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/ExternalCatalogUtils.scala

```diff
@@ -123,6 +124,38 @@ object ExternalCatalogUtils {
     }
     escapePathName(col) + "=" + partitionString
   }
+
+  def prunePartitionsByFilter(
```

Comment:
It is used by the `catalyst/catalog/InMemoryCatalog.scala`.  I picked this up from cherry picking your commit, which introduced other issues.  I may have unnecessarily complicated it.  Let me try and remove this and get back to you.  Thanks.